patches/api/0351-Block-Ticking-API.patch

```diff
@@ -0,0 +1,40 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Sun, 26 Dec 2021 13:23:52 -0500
+Subject: [PATCH] Block Ticking API
+
+
+diff --git a/src/main/java/org/bukkit/block/Block.java b/src/main/java/org/bukkit/block/Block.java
+index 0dd0beee6800fb34520dfa2d05e5bdda76624d64..6ad910b8d48d890ded565e50fa7b89af3211336b 100644
+--- a/src/main/java/org/bukkit/block/Block.java
++++ b/src/main/java/org/bukkit/block/Block.java
+@@ -540,6 +540,29 @@ public interface Block extends Metadatable, net.kyori.adventure.translation.Tran
+      * @return true if the block was destroyed
+      */
+     boolean breakNaturally(@NotNull ItemStack tool, boolean triggerEffect);
++
++    /**
++     * Causes the block to be ticked, this is different from {@link Block#tickRandomly()},
++     * in that it is usually scheduled to occur, for example
++     * redstone components being activated, sand falling, etc.
++     */
++    void tick();
++
++    /**
++     * Causes the block to be ticked randomly.
++     * This has a chance to execute naturally if {@link Block#isRandomlyTicked()} is true.
++     * <p>
++     * For certain blocks, this behavior may be the same as {@link Block#tick()}.
++     */
++    void tickRandomly();
```

Comment:
I think randomTick would be a better name than tickRandomly, as this is executing the random tick logic for the block, not causing it to tick randomly or something